build-tools/packages/build-tools/src/type-test-generator/typeTestUtils.ts

```diff
@@ -0,0 +1,281 @@
+	packageObject: PackageJson,
+	dependencyName: string,
+): void {
+	const dependencyVersion = packageObject?.devDependencies?.[dependencyName];
+	if (typeof dependencyVersion !== "string") {
+		throw new Error(`Did not find devDependency ${dependencyName} in package.json`);
+	}
+}
+
+/**
+ * Fetches the path of the previous package.json or throws an error if not found.
+ * @param previousBasePath - A string representing the path to the root of a package
+ * @returns
+ */
+export function getPreviousPackageJsonPath(previousBasePath: string): string {
+	const previousPackageJsonPath = path.join(previousBasePath, "package.json");
+	if (!existsSync(previousPackageJsonPath)) {
+		throw new Error(`${previousPackageJsonPath} not found.`);
+	}
+	return previousPackageJsonPath;
+}
+
+/**
+ * Attempts to retrieve  a specified type of rollup file path for type definitions from the API Extractor configuration.
+ * @param {string} rollupType - The type of rollup file path to retrieve (ex: "alpha", "beta", "public").
+ * @returns {string} The path to the type definitions file for the specified rollupType, or undefined it cannot be found.
+ * @throws {Error} If api-extractor config cannot be loaded.
+ */
+export function getTypeRollupPathFromExtractorConfig(
+	rollupType: "alpha" | "beta" | "public" | "untrimmed",
+	previousBasePath: string,
+): string | undefined {
+	try {
+		//Load the api-extractor
+		const extractorConfigOptions = ExtractorConfig.tryLoadForFolder({
+			startingFolder: previousBasePath,
+		});
+		if (!extractorConfigOptions || !extractorConfigOptions.configObject) {
+			console.warn(
+				"API Extractor configuration not found. Falling back to default behavior.",
+			);
+			return undefined;
+		}
+		const apiExtractorConfig = extractorConfigOptions.configObject;
+		if (apiExtractorConfig.dtsRollup) {
+			let rollupPath: string | undefined;
+			if (rollupType === "untrimmed") {
+				rollupPath = apiExtractorConfig.dtsRollup.untrimmedFilePath;
+			} else {
+				rollupPath = apiExtractorConfig.dtsRollup[`${rollupType}TrimmedFilePath`];
+			}
+			if (!rollupPath) {
+				console.warn(`Rollup path for "${rollupType}" not found.`);
+				return undefined;
+			}
+			return rollupPath;
+		} else {
+			console.warn(`dtsRollup configuration not found in the API Extractor configuration.`);
+			return undefined;
+		}
+	} catch (error) {
+		console.error(`Error loading API Extractor configuration: ${error}`);
+		throw error;
+	}
+}
+
+/**
+ * Attempts to extract the type definition file path from the 'exports' field of a given package.json.
+ * Checks both 'import' and 'require' resolution methods to find the appropriate path.
+ * If the path is found, it is returned. Otherwise, an error is thrown.
+ * @param previousPackageJson
+ * @returns string - A type definition filepath based on the appropriate export.
+ */
+function getTypePathFromExport(previousPackageJson: PackageJson, previousBasePath: string): string {
+	if (!previousPackageJson.exports) {
```

Comment:
`getTypeRollupPathFromExtractorConfig`, which serves a similar role, returns `undefined` and logs warnings when the relevant configuration options aren't found, but this one throws an exception. I think it would be nice for the two functions to behave similarly.

It would probably be best for this to return `undefined` in cases where config options are missing too. That way the caller can fall back to some default as needed.